### PR TITLE
How to pass arguments to ActiveJob Jobs

### PR DESCRIPTION
**Pull request by dhiachou**
A section explaining how to pass arguments to Jobs has been added.

### CODE REVIEW

**Comment by repinel** [comment]
@embedMaster Could you add [ci skip] to the commit message?

So we don't loose track, fixes #21073

**Comment by dhiachou** [comment]
@repinel  Commit message fixed.

File: guides/source/active_job_basics.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ After reading this guide, you will know:
 
 * How to create jobs.
 * How to enqueue jobs.
+* How to pass arguments to a jobs.
```
Comment (robin850) [inline guides/source/active_job_basics.md]: :scissors: ; there's no need to list all the things we cover in the guide, just the main points!

Comment (dhiachou) [inline guides/source/active_job_basics.md]: Oh I there is a typo there too, I missed it. 
Do you suggest I remove the whole line ?

Comment (repinel) [inline guides/source/active_job_basics.md]: Yes

**Comment by robin850** [comment]
Hi @dhiachou,

Thanks for your patch! What do you think about improving the "Enqueue the Job" part rather than creating a new one ? The examples could use `GuestsCleanupJob` rather than `MyJob` and you could add another example under the first one like:

~~~ruby
# `perform_now` and `perform_later` will call `perform` under the hood so
# you can pass as many arguments as defined in the latter.
GuestsCleanupJob.perform_later(guest1, guest2)
~~~

**Comment by dhiachou** [comment]
Hi @robin850 ,
I believe it is better to emphasize that however ```perform``` is defined, ```perform_later``` and ```perform_now``` take the same args. The example above uses the splat *args. It may be understood that the only way to define is that way.

**Comment by dhiachou** [comment]
@robin850 Please check the last commit

**Comment by robin850** [comment]
@dhiachou : Thanks for the quick update! I see your point regarding `*args` ; maybe we can rename it `*guests` ? I still think that we should remove the "Pass arguments to a Job" part though.

**Comment by dhiachou** [comment]
@robin850 : I merged both sections, and illustrated it with an example using normal parameters, splat parameters and double splat parameters for more clarity.

File: guides/source/active_job_basics.md [hunk 2]
```diff
@@ -70,12 +70,19 @@ Here's what a job looks like:
...
   queue_as :default
 
-  def perform(*args)
+  def perform(*guests)
     # Do something later
   end
 end
 ```
 
+Note that you can define ```perform``` with as many arguments you want. For example:
```
Comment (robin850) [inline guides/source/active_job_basics.md]: Can you use simple backticks around `perform` please ?

File: guides/source/active_job_basics.md [hunk 3]
```diff
@@ -70,12 +70,19 @@ Here's what a job looks like:
...
 
-  def perform(*args)
+  def perform(*guests)
     # Do something later
   end
 end
 ```
 
+Note that you can define ```perform``` with as many arguments you want. For example: 
+```ruby
```
Comment (robin850) [inline guides/source/active_job_basics.md]: Could you remove this example please ? The sentence should be enough.

**Comment by robin850** [comment]
Sorry for the late answer here ; there are just two comments left and this should be good to go!

**Comment by dhiachou** [comment]
@robin850  Please check the new commit

File: guides/source/active_job_basics.md [hunk 4]
```diff
@@ -70,12 +70,14 @@ Here's what a job looks like:
...
   queue_as :default
 
-  def perform(*args)
+  def perform(*guests)
     # Do something later
   end
 end
 ```
 
+Note that you can define `perform` with as many arguments you want.
```
Comment (robin850) [inline guides/source/active_job_basics.md]: There's just a missing "as" before you.

Could you also squash your commits together please ? Then it should be good to go! :-)

**Comment by dhiachou** [comment]
How to squash them together ?

**Comment by dhiachou** [comment]
All set

**Comment by robin850** [comment]
Thank you very much! :-)

**Comment by dhiachou** [comment]
Is it possible to port back to Rails  4 docs too?

**Comment by robin850** [comment]
@dhiachou : Yes sure, sorry for the late answer &mdash; this has been backported as https://github.com/rails/rails/commit/7974611ee7df22174175b59094bfa5c971302efa.

**Pull request merged**